### DISCUSSION TITLE
How does Airflow re-direct print statements to be log messages?

### DISCUSSION THREAD

**Discussion by ryanherring** [apache/airflow#38509] [General]
I'd like to better understand where the logic to re-direct print statements to be log messages is in the Airflow code. I'd like to implement custom formatting for print statements that is different from calls to `logging.info`, `logging.debug`, etc.

**Comment by potiuk** [comment]
I think that requires close studying of https://github.com/apache/airflow/tree/main/airflow/utils/log . It's not documented in detail, but it's a good idea, for someone who studies it, to document it back as contribution. That would be cool.
